### PR TITLE
Incorporation of Repoquery into DNF

### PR DESCRIPTION
**Pull request by j-mracek**
Also documentation was removed and modified to keep track.

### CODE REVIEW

**Comment by j-mracek** [comment]
rpm-software-management/dnf-plugins-core#505

File: tests/test_download.py [hunk 1]
```diff
@@ -53,6 +52,30 @@ def evr(self):
...
+
+    @property
+    def source_name(self):
+        """"
+        returns name of source package
+        e.g. krb5-libs -> krb5
+        """
+        if self.sourcerpm is not None:
+            # trim suffix first
+            srcname = dnf.util.rtrim(self.sourcerpm, ".src.rpm")
```
Comment (ignatenkobrain) [inline tests/test_download.py]: btw in some cases there can be also `.nosrc.rpm`

Comment (j-mracek) [inline tests/test_download.py]: @ignatenkobrain Ok, please can you provide some more information and real example where it fails.

Comment (ignatenkobrain) [inline tests/test_download.py]: http://developers.redhat.com/blog/2014/12/10/how-to-package-proprietary-software/

If you put `NoSource: X` into spec file - rpmbuild will create `.nosrc.rpm`. It's not specific to your PR, just made note about thing.

**Comment by MichaelMraka** [comment]
Merged
67d77a1631724a2a8aaae4a2a5bdcd40679d4cc6

**Pull request closed without merging**